src/main/java/banana/Duke.java

```diff
@@ -0,0 +1,352 @@
+                    text += "T ~ " + doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + "\n";
+                } else if (tasks.getTask(i) instanceof Deadline) {
+                    Deadline dl = (Deadline) tasks.getTask(i);
+                    text += "D ~ " + doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + " ~ " + dl.getDeadLine() + "\n";
+                } else if (tasks.getTask(i) instanceof Event) {
+                    Event ev = (Event) tasks.getTask(i);
+                    text += "E ~ " + doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + " ~ " + ev.getEvent() + "\n";
+                } else {
+                    text += doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + "\n";
+                }
+            }
+        }
+        fw.write(text);
+        fw.close();
+    }
+
+}
+
+/**
+ * The DukeException class
+ * throws specialised Duke exceptions.
+ *
+ * @author: Ravi Ananya
+ **/
+class DukeException extends Exception {
+
+    private String text;
+
+    /**
+     * Constructor for DukeException.
+     *
+     * @param text user input.
+     */
+    public DukeException(String text) {
+        this.text = text;
+    }
+
+    /**
+     * Gets the error message.
+     *
+     * @return the error message.
+     */
+    @Override
+    public String getMessage() {
+        return text;
+    }
+
+}
+
+/**
+ * The Task class handles tasks.
+ *
+ * @author: Ravi Ananya
+ **/
+class Task {
+
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description user input.
+     */
+    public Task(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Gets the string notation of done
+     * or not done.
+     *
+     * @return x for done or empty for not done.
+     */
+    public String getIsDone() {
+        return isDone ? "[X]" : "[ ]";
+    }
+
+    /**
+     * Gets the user input.
+     *
+     * @return the user input.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Sets the task as done.
+     */
+    public void setIsDone() {
+        isDone = true;
+    }
+
+    @Override
+    public String toString() {
+        return getIsDone() + " " + description;
+    }
+
+}
+
+/**
+ * This class handles to-do
+ * types of tasks.
+ *
+ * @author: Ravi Ananya
+ **/
+
+class ToDo extends Task {
+
+    /**
+     * Constructor for ToDo.
+     *
+     * @param description user input.
+     */
+    public ToDo(String description) {
+        super(description);
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+
+}
+
+/**
+ * This class handles deadline
+ * types of tasks.
+ *
+ * @author: Ravi Ananya
+ **/
+
+class Deadline extends Task {
+
+    protected String deadLine;
```

Comment:
Should this variable be renamed to **deadline** instead, since deadline is a single word and naming it deadLine may cause confusion?